x/mongo/driver/topology/topology.go

```diff
@@ -444,9 +440,9 @@ func (t *Topology) selectServerFromSubscription(ctx context.Context, subscriptio
 	for {
 		select {
 		case <-ctx.Done():
-			return nil, ctx.Err()
+			return nil, ServerSelectionError{Wrapped: ctx.Err(), Desc: t.Description()}
```

Comment:
I'm not 100% sure about this, but instinctively, I think we should be using the last seen topology description in the error message rather than fetching the current one from `t.Description()`. However, there's a chance that this is the first loop iteration so `current` hasn't been initialized.

What do you think about initializing `current` above the loop as `current := t.Description()` and then doing `return nil, ServerSelectionError{Wrapped: ctx.Error(), Desc: current}` here?

Reply:
This makes sense, I've changed to use current as the last seen topology description